### PR TITLE
dcerpc: use vecdeque tx iterator

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5321

Describe changes:
- Use VecDeque tx iterator for dcerpc

Replaces #7408 with id+1

### CODE REVIEW

File: rust/src/dcerpc/dcerpc.rs [hunk 1]
```diff
@@ -186,6 +186,13 @@ pub struct DCERPCTransaction {
     pub tx_data: AppLayerTxData,
 }
 
+impl Transaction for DCERPCTransaction {
+    fn id(&self) -> u64 {
+        // need +1 to match state.tx_id
+        self.id + 1
```
Comment (jasonish) [inline rust/src/dcerpc/dcerpc.rs:192]: Is this correct? Other parsers aren't adding 1, whats the difference with this parser?

Comment (jasonish) [reply rust/src/dcerpc/dcerpc.rs:192]: Ah, this one doesn't incremement before assigning. It used to be that a tx_id of 0 cause issues elsewhere so we always started at 1.  Maybe this got fixed elsewhere, or was never an issue for Rust.  I think a comment is in order as its the odd one out.

Comment (catenacyber) [reply rust/src/dcerpc/dcerpc.rs:192]: > Is this correct? Other parsers aren't adding 1, whats the difference with this parser?

The previous PR did not do it and CI fails (make check, and S-V dcerpc)

Is it good for you ?

Comment (jasonish) [reply rust/src/dcerpc/dcerpc.rs:192]: > > Is this correct? Other parsers aren't adding 1, whats the difference with this parser?
> 
> The previous PR did not do it and CI fails (make check, and S-V dcerpc)
> 
> Is it good for you ?

Yes.  Just required some extra review as its different than the others.

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on tlpw1_files_sha256.

ERROR: QA failed on tlpr1_alerts_cmp.

ERROR: QA failed on ips_afp_drop_chk.
  

Pipeline 7460

**Comment by victorjulien** [comment]
Merged in #7423, thanks!

**Pull request closed without merging**